app/src/modules/content/routes/item.vue

```diff
@@ -352,20 +352,30 @@ async function saveVersionAction(action: 'main' | 'stay' | 'quit') {
 				refresh();
 				break;
 			case 'quit':
-				if (!props.singleton) router.push(`/content/${props.collection}`);
+				redirectAfterSave();
 				break;
 		}
 	} catch {
 		// Save version shows unexpected error dialog
 	}
 }
 
+function redirectAfterSave() {
+	const returnToPreviousPage = router.currentRoute.value.query?.return;
+
+	if (returnToPreviousPage === 'true' && !props.singleton) {
```

Comment:
Instead of including `!props.singleton` in each if clause, can we have an early return if it is a singleton so the logic is a bit clearer?